src/libaktualizr/uptane/managedsecondary.cc

```diff
@@ -103,16 +104,29 @@ bool ManagedSecondary::putRoot(const std::string &root, const bool director) {
   return true;
 }
 
-bool ManagedSecondary::sendFirmware(const std::string &data) {
+bool ManagedSecondary::sendFirmwareAsync(const std::string &data) {
+  if (!install_future.valid() || install_future.wait_for(std::chrono::seconds(0)) == std::future_status::ready) {
+    install_future = std::async(std::launch::async, &ManagedSecondary::sendFirmware, this, data);
+    return true;
+  }
+  return false;
+}
+
+bool ManagedSecondary::sendFirmware(std::string data) {  // NOLINT(performance-unnecessary-value-param)
```

Comment:
Why is this passed by value?

Reply:
Because we run sendFirmware in the thread and 'data' is destroyed before we finish installation.

Reply:
Alternatievly we could use shared_ptr.

Reply:
That might be a better idea. Making copies of the firmware blobs is likely to be expensive.